### PR TITLE
Do not add subscriptions marked do_not_invoice to customer invoices

### PR DESCRIPTION
**Pull request by gbova**
https://dimagi-dev.atlassian.net/browse/HI-457

### CODE REVIEW

File: corehq/apps/accounting/invoicing.py [hunk 1]
```diff
@@ -347,12 +347,13 @@ def __init__(self, date_start, date_end, account, recipients=None):
         self.subscriptions = {}
 
     def create_invoice(self):
-        for subscription in self.account.subscription_set.all():
-            if should_create_invoice(subscription, subscription.subscriber.domain, self.date_start, self.date_end):
-                if subscription.plan_version in self.subscriptions:
-                    self.subscriptions[subscription.plan_version].append(subscription)
+        for sub in self.account.subscription_set.all():
```
Comment (nickpell) [inline corehq/apps/accounting/invoicing.py]: nit: adding a filter to subscription_set would avoid adding complexity to the if statement.

File: corehq/apps/accounting/invoicing.py [hunk 2]
```diff
@@ -347,12 +347,13 @@ def __init__(self, date_start, date_end, account, recipients=None):
...
+            if not sub.do_not_invoice \
```
Comment (nickpell) [inline corehq/apps/accounting/invoicing.py]: Just checking: is ```do_not_invoice``` the correct field to consider?  For example, there could be community subscriptions with ```do_not_invoice == False```.  Do we want to bill for users there?

Comment (gbova) [reply corehq/apps/accounting/invoicing.py]: That's a really good question. I assumed that we would still want to bill for community subscriptions on customer enterprise accounts. But if we always want to ignore all community subscriptions, then this isn't the right approach

Comment (nickpell) [reply corehq/apps/accounting/invoicing.py]: I'm not sure what Julie had in mind, should confirm that we're all on the same page.

File: corehq/apps/accounting/invoicing.py [hunk 3]
```diff
@@ -347,12 +347,13 @@ def __init__(self, date_start, date_end, account, recipients=None):
...
+                    and should_create_invoice(sub, sub.subscriber.domain, self.date_start, self.date_end):
+                if sub.plan_version in self.subscriptions:
```
Comment (kaapstorm) [inline corehq/apps/accounting/invoicing.py]: Totally optional, non-blocking nit, and I know you didn't write this ... but if line 347 was
```python
    self.subscriptions = defaultdict(list)
```
then lines 353 to 356 could be replaced with 
```python
    self.subscriptions[sub.plan_version].append(sub)
```

**Review by nickpell — APPROVED** [review]

**Pull request merged**